org-code-javabuilder/lib/src/main/java/dev/javabuilder/LocalContentManager.java

```diff
@@ -0,0 +1,94 @@
+package dev.javabuilder;
+
+import static dev.javabuilder.LocalWebserverConstants.DIRECTORY;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.code.javabuilder.*;
+import org.code.protocol.ContentManager;
+import org.code.protocol.InternalErrorKey;
+import org.code.protocol.JavabuilderException;
+import org.code.protocol.LoggerUtils;
+import org.json.JSONException;
+
+public class LocalContentManager implements ContentManager, ProjectFileLoader {
+  private static final String SERVER_URL_FORMAT = "http://localhost:8080/%s/%s";
+
+  private ProjectData projectData;
+
+  public LocalContentManager() {
+    this.projectData = null;
+  }
+
+  // Used for testing
+  LocalContentManager(ProjectData projectData) {
+    this.projectData = projectData;
+  }
+
+  @Override
+  public UserProjectFiles loadFiles() throws InternalServerError, UserInitiatedException {
+    this.loadProjectDataIfNeeded();
+    return this.projectData.getSources();
+  }
+
+  @Override
+  public String generateAssetUrl(String filename) {
```

Comment:
Good call - I named it like this to mirror the existing `generateAssetUrl` in GlobalProtocol, but yeah I think it makes more sense to rename it `getAssetUrl` here since it won't be exposed to other classes anyway, but only called via GlobalProtocol for now.